funcx_endpoint/funcx_endpoint/executors/high_throughput/funcx_manager.py

```diff
@@ -417,17 +486,18 @@ def poll_funcx_task_socket(self, test=False):
                 self.worker_map.register_worker(w_id, reg_info['worker_type'])
 
             elif m_type == b'TASK_RET':
+                # We lock because the following steps are also shared by task_cancel
+                self.task_cancel_lock.acquire()
```

Comment:
this is more a lock to do with "result processing" than with cancellation, then?